### ISSUE TITLE
migrate the `akka.japi.Creator` to `java.util.function.Supplier`?

### ISSUE THREAD

**Issue by hepin1989** [akka/akka#25773]
I found the Creator is not clear to use in Java API form the first sense, and the `Supplier` make me feel more natively.

**Comment by patriknw** [comment]
right, but we can't do it without breaking binary backwards compat
same with many of the other functional interfaces in akka.japi

**Comment by raboof** [comment]
Jup we must postpone removing this class and updating existing usages until we can break binary compat, but we could add a note to the scaladoc suggesting preferring `Supplier` for new code already.

**Comment by hepin1989** [comment]
Could we add a new overloaded method and deprecated the `Creator` based one latter and remove them in 3.0?

**Comment by patriknw** [comment]
I don't think overloading will work, when using it with lambdas it will be ambiguous

**Comment by hepin1989** [comment]
> I don't think overloading will work, when using it with lambdas it will be ambiguous

That's true ......

**Comment by vfragosop** [comment]
Wouldn't it be possible to extend native generics, then implement default methods, then replace all interfaces with their native counterpart?

```java
public class Creator<T> extends Supplier<T>
  public T create();
  default T get() {
    return create();
  }
}
```
